CMakeLists.txt

```diff
@@ -0,0 +1,277 @@
+cmake_minimum_required(VERSION 3.5)
+
+project(
+  libmbus
+  LANGUAGES CXX C
+  VERSION "0.9.0")
+
+if(CMAKE_BUILD_TYPE STREQUAL "")
+  message(STATUS "CMAKE_BUILD_TYPE empty setting to Debug")
+  set(CMAKE_BUILD_TYPE "Debug")
+endif()
+
+# ##############################################################################
+# default options -> changed with e.g. cd build && cmake ..
+# -DLIBMBUS_BUILD_TESTS=ON
+# ##############################################################################
+
+option(LIBMBUS_BUILD_EXAMPLES "build examples" OFF)
+option(LIBMBUS_BUILD_TESTS "build tests" OFF)
+option(LIBMBUS_ENABLE_COVERAGE "build with coverage support" OFF)
+option(LIBMBUS_RUN_CLANG_TIDY "use Clang-Tidy for static analysis" OFF)
+option(LIBMBUS_PACKAGE_DEB "build debian package" OFF)
+option(LIBMBUS_PACKAGE_RPM "build rpm package" OFF)
+option(LIBMBUS_BUILD_DOCS "build documentation" OFF)
+
+set(CMAKE_CXX_STANDARD 17)
+set(CMAKE_CXX_STANDARD_REQUIRED ON)
+set(CMAKE_CXX_EXTENSIONS OFF)
+set(CMAKE_C_STANDARD 11)
+
+# Append our module directory to CMake
+list(APPEND CMAKE_MODULE_PATH ${CMAKE_CURRENT_SOURCE_DIR}/cmake)
+list(APPEND CMAKE_MODULE_PATH ${CMAKE_BINARY_DIR})
+
+# Set the output of the libraries and executables.
+set(EXECUTABLE_OUTPUT_PATH ${PROJECT_BINARY_DIR}/bin)
+set(LIBRARY_OUTPUT_PATH ${PROJECT_BINARY_DIR}/lib)
+
+# ##############################################################################
+# static analysis
+# ##############################################################################
+
+if(LIBMBUS_RUN_CLANG_TIDY)
+  find_program(
+    CLANG_TIDY_EXE
+    NAMES "clang-tidy"
+    DOC "/usr/bin/clang-tidy")
+  if(NOT CLANG_TIDY_EXE)
+    message(WARNING "clang-tidy not found.")
+  else()
+    message(STATUS "clang-tidy found: ${CLANG_TIDY_EXE}")
+    set(DO_CLANG_TIDY "${CLANG_TIDY_EXE}")
```

Comment:
Is this `DO_CLANG_TIDY` variable really needed?
It is only used once. Can't it one use be replaced with `CLANG_TIDY_EXE`?